infra/feast-operator/internal/controller/services/repo_config.go

```diff
@@ -44,71 +45,154 @@ func (feast *FeastServices) getServiceFeatureStoreYaml(feastType FeastServiceTyp
-					filePersistence := services.OnlineStore.Persistence.FilePersistence
-					path = getActualPath(filePersistence.Path, filePersistence.PvcConfig)
-				}
-
-				repoConfig.OnlineStore = OnlineStoreConfig{
-					Type: OnlineSqliteConfigType,
-					Path: path,
+				err := setRepoConfigOnline(services, secretExtractionFunc, &repoConfig)
+				if err != nil {
+					return repoConfig, err
 				}
 			}
 		case RegistryFeastType:
 			// Registry server only has a `registry` section
-			if isLocalRegistry {
-				path := DefaultRegistryEphemeralPath
-				var s3AdditionalKwargs *map[string]string
-				if services != nil && services.Registry != nil && services.Registry.Local != nil &&
-					services.Registry.Local.Persistence != nil && services.Registry.Local.Persistence.FilePersistence != nil {
-					filePersistence := services.Registry.Local.Persistence.FilePersistence
-					path = getActualPath(filePersistence.Path, filePersistence.PvcConfig)
-					s3AdditionalKwargs = filePersistence.S3AdditionalKwargs
-				}
-				repoConfig.Registry = RegistryConfig{
-					RegistryType:       RegistryFileConfigType,
-					Path:               path,
-					S3AdditionalKwargs: s3AdditionalKwargs,
+			if isLocalReg {
+				err := setRepoConfigRegistry(services, secretExtractionFunc, &repoConfig)
+				if err != nil {
+					return repoConfig, err
 				}
-				repoConfig.OfflineStore = OfflineStoreConfig{}
-				repoConfig.OnlineStore = OnlineStoreConfig{}
 			}
 		}
 	}
 
 	return repoConfig, nil
 }
 
+func setRepoConfigRegistry(services *feastdevv1alpha1.FeatureStoreServices, secretExtractionFunc func(secretRef string, secretKeyName string) (map[string]interface{}, error), repoConfig *RepoConfig) error {
+	var parametersMap map[string]interface{}
+	filePath := DefaultRegistryEphemeralPath
+	var s3AdditionalKwargs *map[string]string
+	var persistenceType string
+	if services != nil && services.Registry != nil && services.Registry.Local != nil &&
+		services.Registry.Local.Persistence != nil && services.Registry.Local.Persistence.FilePersistence != nil {
+		persistenceType = string(RegistryFileConfigType)
+		filePersistence := services.Registry.Local.Persistence.FilePersistence
+		filePath = getActualPath(filePersistence.Path, filePersistence.PvcConfig)
+		s3AdditionalKwargs = filePersistence.S3AdditionalKwargs
+	} else if services != nil && services.Registry != nil && services.Registry.Local != nil &&
+		services.Registry.Local.Persistence != nil && services.Registry.Local.Persistence.DBPersistence != nil {
+		if len(services.Registry.Local.Persistence.DBPersistence.Type) > 0 {
+			var err error
+			filePath = ""
+			persistenceType = services.Registry.Local.Persistence.DBPersistence.Type
+			secretKeyName := services.Registry.Local.Persistence.DBPersistence.SecretKeyName
+			if len(secretKeyName) == 0 {
+				secretKeyName = persistenceType
+			}
+			parametersMap, err = secretExtractionFunc(services.Registry.Local.Persistence.DBPersistence.SecretRef.Name, secretKeyName)
```

Comment:
no need to declare an error var... instead use a short assignment statement as we do elsewhere in the code.
```suggestion

			filePath = ""
			persistenceType = services.Registry.Local.Persistence.DBPersistence.Type
			secretKeyName := services.Registry.Local.Persistence.DBPersistence.SecretKeyName
			if len(secretKeyName) == 0 {
				secretKeyName = persistenceType
			}
			parametersMap, err := secretExtractionFunc(services.Registry.Local.Persistence.DBPersistence.SecretRef.Name, secretKeyName)
```

Reply:
I can't since parametersMap is defined outside the if and is used outside the if and if I use `:=` it shadows that value
```	
repoConfig.Registry = RegistryConfig{
		RegistryType:       RegistryConfigType(persistenceType),
		Path:               filePath,
		S3AdditionalKwargs: s3AdditionalKwargs,
		DBParameters:       parametersMap,
}
```

Reply:
> if I use := it shadows that value
what does this mean? 
fwiw, go will detect that `parametersMap` has already been declared and reassign it accordingly.

e.g. this will print `3 4 5 6`
```go
package main

import "fmt"

func main() {
	var i, j int = 1, 2
	k := 3
	j, k, l := 4, 5, 6

	fmt.Println(i, j, k, l)
}
```

Reply:
When I change that to `:=` I am getting an error `unused variable parametersMap`

Reply:
ok... because its not being used until after the if statement... i see now.